djmoney/contrib/exchange/models.py

```diff
@@ -64,17 +55,20 @@ def get_rate(source, target, backend=None):
 def _get_rate(source, target, backend):
     if text_type(source) == text_type(target):
         return 1
-    try:
-        forward = models.Q(currency=target, backend__base_currency=source)
-        reverse = models.Q(currency=source, backend__base_currency=target)
-        return Rate.objects.annotate(
-            rate=models.Case(
-                models.When(forward, then=models.F('value')),
-                models.When(reverse, then=models.Value(get_one()) / models.F('value')),
-            )
-        ).get(forward | reverse, backend=backend).rate
-    except Rate.DoesNotExist:
+    rates = Rate.objects.filter(currency__in=(source, target), backend=backend)
+    if not rates:
         raise MissingRate('Rate %s -> %s does not exist' % (source, target))
+    # Direct rate
+    if len(rates) == 1:
+        rate = rates[0]
+        if text_type(rate.currency) == text_type(target):
+            return rate.value
+        return 1 / rate.value
+    # Indirect rate
+    first, second = rates
+    if text_type(first.currency) == text_type(target):
+        first, second = second, first
```

Comment:
Wondering: Why this swap?

Reply:
There is no guarantee, that the queryset will be ordered somehow, so the first element could be for target currency or for source currency. So, instead of adding an expression to `order_by`, that will put the source currency in the first place I decided to add an extra check here and swap items if they are ordered not as expected